contracts/transfer/src/tree.rs

```diff
@@ -26,7 +28,14 @@ impl Tree {
         }
     }
 
+    /// Push one [`NoteLeaf`] onto the [`Tree`], filtering out notes that are
+    /// transparent and have a value of 0.
     pub fn push(&mut self, mut leaf: NoteLeaf) -> Note {
+        // skip transparent notes with a value of 0
+        if leaf.note.value(None).is_ok_and(|value| value == 0) {
+            return leaf.note;
+        }
```

Comment:
Should this return an Option?

Reply:
I had a similar idea but didn't want to change the contract too much